test/access/AccessControl.behavior.js

```diff
@@ -210,8 +211,293 @@ function shouldBehaveLikeAccessControlEnumerable(errorPrefix, admin, authorized,
+      const firstDefaultAdminTransferDelayedUntil = (await time.latest()).add(delay);
+      expect(await this.accessControl.pendingDefaultAdmin()).to.equal(newDefaultAdmin);
+      expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(
+        firstDefaultAdminTransferDelayedUntil,
+      );
+
+      // Time passes just before delay
+      await time.increaseTo(firstDefaultAdminTransferDelayedUntil.subn(1));
+
+      // defaultAdmin changes its mind and begin again to another address
+      await this.accessControl.beginDefaultAdminTransfer(other, { from: defaultAdmin });
+      const secondDefaultAdminTransferDelayedUntil = (await time.latest()).add(delay);
+      expect(await this.accessControl.pendingDefaultAdmin()).to.equal(other);
+      expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(
+        secondDefaultAdminTransferDelayedUntil,
+      );
+    });
+
+    it('should be able to begin a transfer again after delay pass if not accepted', async function () {
+      // First defaultAdmin transfer started
+      await this.accessControl.beginDefaultAdminTransfer(newDefaultAdmin, { from: defaultAdmin });
+      const firstDefaultAdminTransferDelayedUntil = (await time.latest()).add(delay);
+      expect(await this.accessControl.pendingDefaultAdmin()).to.equal(newDefaultAdmin);
+      expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(
+        firstDefaultAdminTransferDelayedUntil,
+      );
+
+      // Time passes after delay without acceptance
+      await time.increaseTo(firstDefaultAdminTransferDelayedUntil.addn(1));
+
+      // defaultAdmin changes its mind and begin again to another address
+      await this.accessControl.beginDefaultAdminTransfer(other, { from: defaultAdmin });
+      const secondDefaultAdminTransferDelayedUntil = (await time.latest()).add(delay);
+      expect(await this.accessControl.pendingDefaultAdmin()).to.equal(other);
+      expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(
+        secondDefaultAdminTransferDelayedUntil,
+      );
+    });
+
+    it('should revert if it is called by non-admin accounts', async function () {
+      await expectRevert(
+        this.accessControl.beginDefaultAdminTransfer(newDefaultAdmin, { from: other }),
+        `${errorPrefix}: account ${other.toLowerCase()} is missing role ${DEFAULT_ADMIN_ROLE}`,
+      );
+    });
+  });
+
+  describe('accepts transfer admin', async function () {
+    let correctPendingDefaultAdmin;
+    let correctIncreaseTo;
+
+    beforeEach(async function () {
+      // Set as correct so they're explicitly needed in revert tests, since conditions
+      // are not mutually exclusive and accidentally an incorrect value will make it revert
+      // possiblty creating false positives (eg. expect revert for incorrect caller but getting it
+      // from a badly expected delayed until)
```

Comment:
I don't understand what this is saying. Can you rephrase it?

Reply:
I double-checked and I think it's not needed anymore so I'm removing it at https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/880ac0b876437d1618b81ff8785f0233901517c2

Still, I have a question, so let me explain my reasoning before asking:

Currently we have 2 revert strings in `acceptDefaultAdminTransfer`:

- `AccessControl: pending admin must accept`
- `AccessControl: transfer delay not passed`

Those two come from different conditions that were previously merged in the same require:

```solidity
require(isPendingDefaultAdmin() && delayHasPassed(), "AccessControl: Only one revert string for both");
```

The reason why I initially set this comment was because I was explicitly renaming `newDefaultAdmin` to `correctDefaultAdmin` to be clear about the fact that inside the `accepts transfer admin` describe block, the `newDefaultAdmin` is **the** `correctDefaultAdmin`. This is relevant because we had tests like this:

```javascript
it('should revert if block.timestamp is less than delayed until', async function () {
  await time.increaseTo(delayNotPassed.subn(1));
  await expectRevert(
    this.accessControl.acceptDefaultAdminTransfer({ from: newDefaultAdmin }),
    `${errorPrefix}: transfer delay not passed`,
  );
});
```

The reasoning was that it's easier to make a mistake if it reads `newDefaultAdmin` and someone overrode it accidentally in the `before` function. The function would start to throwing the same error without nobody noticing it was because of an incorrect `defaultAdmin`, not because of a delay not yet passed.

I initially didn't split the `require(a() && b(), "a && b")` into `require(a(), "a")` and `require(b(), "b")` because I know it's costly to keep two revert strings.

My question is:
What do we usually prefer? More revert strings for better clarity or less for gas savings?

Reply:
> What do we usually prefer? More revert strings for better clarity or less for gas savings?

I don't have a good answer for this. We have to find some good point in the middle.

Regarding the variable name, I sort of see your point, but I'm not as concerned about it.